### PR TITLE
Refactor vertx-junit5 to host more submodules - Added Vert.x JUnit 5 Web Client

### PR DESCRIPTION
**Pull request by slinkydeveloper**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,6 +15,12 @@
   ~ limitations under the License.
   --><project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
+  <packaging>pom</packaging>
```
Comment (jponge) [inline pom.xml]: I would move the dependencies listed in this POM to `vertx-junit5` (and the ones needed in the other module if need be).

You may also need `dependencyManagement` if you have shared dependencies and you want to factor dependencies.

File: vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/VertxWebClientExtension.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+    Class<?> type = parameterContext.getParameter().getType();
+    return type.equals(WebClient.class) || type.equals(io.vertx.reactivex.ext.web.client.WebClient.class) || type.equals(io.vertx.rxjava.ext.web.client.WebClient.class);
+  }
+
+  @Override
+  public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+    Class<?> type = parameterContext.getParameter().getType();
+    ExtensionContext.Store store = extensionContext.getStore(ExtensionContext.Namespace.create(VertxWebClientExtension.class, extensionContext));
+    if (WebClient.class.equals(type))
```
Comment (jponge) [inline vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/VertxWebClientExtension.java]: There and everywhere, please have blocks even for single statements. Saves your life one day 😉

File: vertx-junit5-web-client/src/test/java/io/vertx/junit5/web/TestRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package io.vertx.junit5.web;
```
Comment (jponge) [inline vertx-junit5-web-client/src/test/java/io/vertx/junit5/web/TestRequestTest.java]: You can have license headers here as well.

File: vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/VertxWebClientExtension.java (hunk 2 above)
Comment (slinkydeveloper) [reply vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/VertxWebClientExtension.java]: mhhh this battle is even worse than tabs vs spaces :rofl:

Comment (jponge) [reply vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/VertxWebClientExtension.java]: https://blog.codecentric.de/en/2014/02/curly-braces/

File: pom.xml (hunk 1 above)
Comment (slinkydeveloper) [reply pom.xml]: I've moved in parent pom only dependencies shared between modules, this is what you mean?

**Comment by jponge** [comment]
@slinkydeveloper Just added consistent naming in modules.

**Comment by slinkydeveloper** [comment]
Do you want to port these new apis in vertx 3.8?

**Comment by jponge** [comment]
Yes the sooner the better :-)

**Pull request merged**